src/components/documents/CmsText/WithCmsText.tsx

```diff
@@ -2,15 +2,17 @@ import { CmsTextContext, useFirstCmsTextDocumentIfExists } from '@tdev-component
 import { observer } from 'mobx-react-lite';
 
 interface Props {
-    id: string;
+    entries: { [key: string]: string };
     children?: React.ReactNode;
 }
 
-const WithCmsText = observer(({ id, children }: Props) => {
-    const doc = useFirstCmsTextDocumentIfExists(id);
+const WithCmsText = observer(({ entries, children }: Props) => {
+    const allDocumentsAvailable = Object.values(entries).every(
+        (documentRootId) => !!useFirstCmsTextDocumentIfExists(documentRootId)
+    );
```

Comment:
I think this code is problematic - `.every` only proceeds until an entry does not meet the condition for the first time:

```js
[1, 2, 3, 4, 5].every((num) => {
    console.log(num);
    return num < 2;
});
// => 1, 2
// 3, 4, 5 are not processed
```

So, when having more than one key, the `useFirstCmsTextDocumentIfExists` will be called for the first key and only if this one is loaded, the next one will load what leads to a request-waterfall and pretty sure to unstable calls of react hooks...

With the following code, this should be fixed:

```suggestion
    const allDocumentsAvailable = Object.values(entries).map(
        (documentRootId) => !!useFirstCmsTextDocumentIfExists(documentRootId)
    ).every(Boolean);
```

Reply:
Ah yes, I see your point - good catch! I will commit this suggestion and make sure it works :).